os/board/common/partitions.c

```diff
@@ -121,17 +123,21 @@ void configure_partitions(void)
 		}
 
 #if defined(CONFIG_MTD_PARTITION_NAMES)
-		if (*names) {
-			mtd_setpartitionname(mtd_part, names);
-		}
 
-		while (*names) {
-			if (*names == ',') {
+		if (*names) {
+			while (*names) {
+				if (*names == ',') {
+					part_name[index] = '\0';
+					index = 0;
+					names++;
+					break;
+				}
+				part_name[index++] = *names;
```

Comment:
You should check max size of string because you use an index with 16 size.

Reply:
I added this logic.